subsys/net/lib/mqtt/net_utils.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2017 Texas Instruments, Inc
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * The following code was lifted from Zephyr's net_app library, to
+ * support the mqtt port to sockets, and to decouple the mqtt lib from
+ * net_app's struct net_app_ctx, which may evolve with the new
+ * net_pkt/net_buf redesign.
+ * This ideally could move to a generic net/utils module.
+ */
+
+#include <errno.h>
+#include <stdlib.h>
+
+#include "net_utils.h"
+
+static int get_port_number(const char *peer_addr_str,
```

Comment:
There is actually a function called `net_ipaddr_parse()` that parses the IPv[4|6] address + port and returns that info to the caller. Perhaps it could be used here too so we could avoid this util function.